src/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -60,6 +61,18 @@ public ConfidentialClientApplicationBuilder WithCertificate(X509Certificate2 cer
             return this;
         }
 
+        /// <summary>
+        /// Sets the certificate associated with the application
+        /// </summary>
+        /// <param name="certificate">The X509 certificate used as credentials to prove the identity of the application to Azure AD.</param>
+        /// <remarks>You should use certificates with a private key size of at least 2048 bytes. Future versions of this library might reject certificates with smaller keys. </remarks>
+        public ConfidentialClientApplicationBuilder WithCertificate(X509Certificate2 certificate, Dictionary<string, string> claimsToSign)
```

Comment:
Why is claims a dictionary instead of a set? Also, prefer to use interfaces over collections, not actual collections. IDictionary / ISet.

Reply:
This is according to the provided spec. @henrik-me  @jmprieur and thoughts on this?

Reply:
Whether to use IDictionaty, or what is the question?

---
In reply to: [293914940](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1208#discussion_r293914940) [](ancestors = 293914940)

Reply:
That is the question. And the answer is - use IDictionary :)